packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -651,6 +651,9 @@ interface InstanceConfig {
  */
 function createInstances(cluster: DatabaseClusterNew, props: DatabaseClusterBaseProps, subnetGroup: ISubnetGroup): InstanceConfig {
   const instanceCount = props.instances != null ? props.instances : 2;
+  if (Token.isUnresolved(instanceCount)) {
+    throw new Error('Instance count is not a token');
```

Comment:
This error message needs to be better here. Maybe something like:

```suggestion
    throw new Error('The number of instances an RDS Cluster consists of cannot be provided as a deploy-time only value!');
```